api/server.go

```diff
@@ -236,6 +237,58 @@ func (s *server) consensusUpdatesIndexHandler(jc jape.Context) {
 	jc.Encode(res)
 }
 
+func (s *server) miningGetBlockTemplateHandler(jc jape.Context) {
+	var req MiningGetBlockTemplateRequest
+	if jc.Decode(&req) != nil {
+		return
+	}
+
+	// TODO: add polling
+
+	template, err := generateBlockTemplate(s.cm, req.PayoutAddress)
```

Comment:
Do we want to 400 on a void payout address?

Reply:
Probably doesn't hurt.